apache/superset
# build: use manifest hooks for dev server proxy and fix hot reload for charts

### CATEGORY


### SUMMARY

This updates how `wepback-dev-server` proxy reads asset manifest. Instead of reading from the generated `manifest.json` file, we use webpack hooks.

This is needed because sometimes proxy server will read `manifest.json` before it is fully updated, resulting the proxy to return the old code for an updated module on a hard page refresh.

Also added some docs for how to edit plugins as a followup to https://github.com/apache/incubator-superset/pull/9326 . 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A

### TEST PLAN

Before the change:

- Start the dev server, go to any page
- Edit a React component in that page, hot load should work, you component is updated
- Refresh the page, you component is served with the old code

After the change:
- You component should reflect the updated code.

### ADDITIONAL INFORMATION

N/A

### REVIEWERS

@rusackas @kristw
